locked-asset/token-unstake/src/lib.rs

```diff
@@ -0,0 +1,84 @@
+#![no_std]
+
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+pub const DEFAULT_UNBOND_EPOCHS: u64 = 10;
+
+#[derive(ManagedVecItem, TopEncode, TopDecode, NestedEncode, NestedDecode, TypeAbi, Clone)]
+pub struct UnstakePair<M: ManagedTypeApi> {
+    pub unlock_epoch: u64,
+    pub token_payment: EsdtTokenPayment<M>,
+}
+
+#[elrond_wasm::contract]
+pub trait TokenUnstakeModule: token_send::TokenSendModule {
+    #[init]
+    fn init(&self) {
+        self.unbond_epochs().set_if_empty(DEFAULT_UNBOND_EPOCHS);
+    }
+
+    #[only_owner]
+    #[endpoint(setUnbondEpochs)]
+    fn set_unbond_epochs(&self, unbond_epochs: u64) {
+        self.unbond_epochs().set(unbond_epochs);
+    }
+
+    #[payable("*")]
+    #[endpoint(depositUserTokens)]
+    fn deposit_user_tokens(&self, caller: &ManagedAddress) {
+        let token_payment = self.call_value().single_esdt();
+        require!(
+            token_payment.token_nonce == 0,
+            "Can only unstake fungible tokens"
+        );
+        require!(
+            token_payment.amount > 0,
+            "Payment amount must be greater than 0"
+        );
```

Comment:
Not required, since transferring ESDT with 0 amount is no longer possible.

Reply:
Done.